### PR TITLE
Added tests for Write

### PR DESCRIPTION
**Pull request by fgimenez**
First part of changes to address #11584, I'll continue with StdCopy coverage next.

### CODE REVIEW

File: pkg/stdcopy/stdcopy_test.go [hunk 1]
```diff
@@ -49,6 +50,34 @@ func TestWrite(t *testing.T) {
...
+func (f *errWriter) Write(buf []byte) (int, error) {
+	return f.n, f.err
+}
+
+func TestWriteWithWriterError(t *testing.T) {
+	expectedError := errors.New("expected")
+	writer := NewStdWriter(&errWriter{err: expectedError}, Stdout)
+	data := []byte("This won't get written, sigh")
+	_, err := writer.Write(data)
+	if err != expectedError {
```
Comment (vdemeester) [inline pkg/stdcopy/stdcopy_test.go]: To be *complete* we should check the number of byte written (https://github.com/docker/docker/blob/master/pkg/stdcopy/stdcopy.go#L44) too I think :wink:.

Comment (fgimenez) [inline pkg/stdcopy/stdcopy_test.go]: @vdemeester thanks for that :) I'll push the changes shortly

**Comment by runcom** [comment]
LGTM

**Comment by ehazlett** [comment]
+1 for a more detailed commit msg (`git commit --amend`).  otherwise lgtm

**Comment by fgimenez** [comment]
@runcom @ehazlett done, let me know what do you think :)

**Comment by vdemeester** [comment]
@fgimenez once janky is happy, it's gonna be merged :wink:

**Pull request merged**